From c629acdc74c5b775c5cf7a7a3cb20ac833bcd0cc Mon Sep 17 00:00:00 2001
Message-Id: <c629acdc74c5b775c5cf7a7a3cb20ac833bcd0cc.1346940159.git.minovotn@redhat.com>
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 16 Aug 2012 11:39:03 +0200
Subject: [PATCH 01/18] uhci: zap uhci_pre_save

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1345117160-21046-2-git-send-email-kraxel@redhat.com>
Patchwork-id: 40922
O-Subject: [RHEL-6.4 qemu-kvm PATCH 01/18] uhci: zap uhci_pre_save
Bugzilla: 805172
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

Cancel transactions before saving vmstate is pretty pointless and just
causes disruptions.  We need to cancel them before *loading* vmstate,
but in that case uhci_reset() handles it already and no special action
is needed.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
(cherry picked from commit 5852d3bfe16c85d4dabc4e0d21658fc680a756e5)
---
 hw/usb-uhci.c |    8 --------
 1 files changed, 0 insertions(+), 8 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/usb-uhci.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/hw/usb-uhci.c b/hw/usb-uhci.c
index 001a044..9ab4154 100644
--- a/hw/usb-uhci.c
+++ b/hw/usb-uhci.c
@@ -347,13 +347,6 @@ static void uhci_reset(void *opaque)
     uhci_async_cancel_all(s);
 }
 
-static void uhci_pre_save(void *opaque)
-{
-    UHCIState *s = opaque;
-
-    uhci_async_cancel_all(s);
-}
-
 static int uhci_post_load(void *opaque, int version_id)
 {
     UHCIState *s = opaque;
@@ -379,7 +372,6 @@ static const VMStateDescription vmstate_uhci = {
     .version_id = 1,
     .minimum_version_id = 1,
     .minimum_version_id_old = 1,
-    .pre_save = uhci_pre_save,
     .post_load = uhci_post_load,
     .fields      = (VMStateField []) {
         VMSTATE_PCI_DEVICE(dev, UHCIState),
-- 
1.7.11.4

